crates/subspace-farmer/src/bin/subspace-farmer/main.rs

```diff
@@ -41,6 +41,34 @@ enum IdentityCommand {
     },
 }
 
+/// Arguments for farmer
+#[derive(Debug, Parser)]
+struct FarmingArgs {
+    /// Multiaddrs of bootstrap nodes to connect to on startup, multiple are supported
+    #[clap(long)]
+    bootstrap_nodes: Vec<Multiaddr>,
+    /// Custom path for data storage instead of platform-specific default
+    #[clap(long, value_hint = ValueHint::FilePath)]
+    custom_path: Option<PathBuf>,
+    /// Multiaddr to listen on for subspace networking, for instance `/ip4/0.0.0.0/tcp/0`,
+    /// multiple are supported, subspace networking is disabled when none specified
+    #[clap(long)]
+    listen_on: Vec<Multiaddr>,
+    /// WebSocket RPC URL of the Subspace node to connect to
+    #[clap(long, value_hint = ValueHint::Url, default_value = "ws://127.0.0.1:9944")]
+    node_rpc_url: String,
+    /// Host and port where built-in WebSocket RPC server should listen for incoming connections
+    #[clap(long, short, default_value = "127.0.0.1:9955")]
+    ws_server_listen_addr: SocketAddr,
+    /// Address for farming rewards
+    #[clap(long, parse(try_from_str = parse_reward_address))]
+    reward_address: Option<PublicKey>,
+    // TODO: Should we require user to always set plot size?
+    /// Maximum plot size in human readable format (e.g., 1K 234M 2G).
```

Comment:
Just a few questions, is the plot size configurable on each farmer run? Any caveats if the user changes the plot size when restarting? If I change the plot size to a smaller number, more pieces will get evicted even I have enough disk space, right?

Reply:
If you change the plot size to a smaller number, pieces will no longer be appended. No piece will get evicted. I guess we can handle this case and remove pieces on a node start in future. Should I add todo stating this?

Reply:
Nice to confirm these cases, let's see how @nazar-pc think about it. Just have a feeling that there are likely some potential problems if the plot size is misconfigured. Given that the data is valuable, we should treat it with great care.

Reply:
Yes, definitely. We should either refuse to start if we see that current plot size is already bigger than specified or handle it gracefully. Either way TODO should be fine for now.